### ISSUE TITLE
Scanpy 1.5 compatibility

### ISSUE THREAD

**Issue by Marius1311** [theislab/cellrank#141]
AnnData >= 0.7 stores graph related data in `adata.obsp`, we should take this into account when reading/writing attributes, while keeping backward compatibility

**Comment by Marius1311** [comment]
Simple need a function which first looks in `.obsp` for the attribute and, if it's not there, looks in `.uns` and issues a warning.

**Comment by Marius1311** [comment]
Started this in #144

**Comment by michalk8** [comment]
So, is this an `scanpy` versioning problem or `anndata` or both? What I'd do is to modify the functions you've created and from the mentioned version(s), default to `obsp` and fallback to `uns` or vice versa.
Question is, how compatible do you want it to be (asking about whether we should default to anything at all)? Cleanest way would be to look only in `obsp` from >= 0.7 and fail if there's nothing (let the user move it). The only benefit for defaulting to something different (`uns` or `obsp`) is that user doesn't have to move the data themself.

**Comment by Marius1311** [comment]
I think it's both `scanpy` and `anndata`: anndata created the new `.obsp` field and scanpy started storing the output of `.pp.neighbors` in `.obsp` from version 1.5, see the changelog [here](https://scanpy.readthedocs.io/en/latest/)

So let me make sure I understand what you are saying: if the anndata version is >= 0.7, we check in `obsp` and we don't default to `.uns`. If the anndata version is <0.7, we just check in `.uns`?

@VolkerBergen , how are you going to handle this in scvelo?

**Comment by VolkerBergen** [comment]
I created a helper function to maintain compatibility with stored data from older `anndata` versions.
```
def get_neighs(adata, mode="distances"):
    return (
        adata.obsp[mode]
        if hasattr(adata, "obsp") and mode in adata.obsp.keys()
        else adata.uns["neighbors"][mode]
    )
```
In scvelo, `distances` and `connectivities` are stored in `obsp` along the same lines as in scanpy, while the `velocity_graph` and all other pairwise annotations will be moved to `obsp` as well in the next release.

As a side note, there's currently some memory issues with `obsp`, so it will only be fully functional with the next anndata release.

**Comment by michalk8** [comment]
@Marius1311 Is `.obsp` only for square-matrix data, like `cell-to-cell` or `gene-to-gene` graphs? If not, I'd store `Lineage` there as well.

**Comment by michalk8** [comment]
Implemented via #226

**Comment by Marius1311** [comment]
`.obsp` is only for graph-based data, which means only `n_cells x n_cells` makes sense as a shape.
